### DISCUSSION TITLE
Pass SSL certificates contents as string instead of path

### DISCUSSION THREAD

**Discussion by nicholas-alonzo** [psycopg/psycopg2#1583] [Q&A]
Hi,

I'm working on a project that involves connecting to Postgres using SSL certificates. I'm using SQLAlchemy and psycopg2 to connect like so:

```
ssl_args = {
    'sslmode': 'verify-ca',
    'sslcert': '/path/to/ssl.crt',
    'sslkey': '/path/to/ssl.key',
    'sslrootcert': '/path/to/root.crt'
}

db_engine = sqlalchemy.create_engine('postgresql+psycopg2://user:password@host:port/dbname, connect_args=ssl_args)
```

I'm now wondering if there's a valid way to pass the contents of the SSL certificates as a string to connect. I tried this below, but SQLAlchemy throws an error that it's not a valid path.

```
ssl_args = {
    'sslmode': 'verify-ca',
    'sslcert': '---------BEGIN....',
    'sslkey': '---------BEGIN....',
    'sslrootcert': '---------BEGIN....',
}

db_engine = sqlalchemy.create_engine('postgresql+psycopg2://user:password@host:port/dbname, connect_args=ssl_args)
```

Any thoughts or ideas would be appreciated, thank you

**Comment by dvarrazzo** [comment]
Psycopg connection string passes the arguments to the libpq. The supported parameters are documented [here](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS).

If SQLAlchemy does its own manipulation is something you have to ask the SQAlchemy project. I assume there is a way to pass-through.

**Reply by nicholas-alonzo** [reply]
I'll ask in their discussions, thank you for your response 👍
